File: insertion_sort/insertion_sort.py
def insertionSort(alist):
    """
    Performs a insertion sort
    :type alist: object
    """
    for index in range(1, len(alist)):
        current_value = alist[index]
        position = index
        while position > 0 and alist[position - 1] > current_value:
            alist[position] = alist[position - 1]
            position -= 1
            alist[position] = current_value


def main():
    alist = [54, 26, 93, 17, 77, 31, 44, 55, 20]
    insertionSort(alist)
    print(alist)


if __name__ == '__main__':
    main()

File: insertion_sort/InsertionSort.cs
using System;

public class InsertionSort
{
    public static void Sort(int[] data)
    {
        int key;
        for(int i = 1; i < data.Length; i++)
        {
            int j  = i - 1;
            key = data[i];
            while (j >= 0 && data[j] > key)
            {
                data[j + 1] = data[j];
                j = j - 1;
            }
            data[j + 1] = key;
        }
    }

    public static void Main()
    {
        int[] arr = new int[] {1000, 45, -45, 121, 47, 45, 65, 121, -1, 103, 45, 34};
        Sort(arr);
        for(int i = 0; i < arr.Length; i++)
        {
            Console.Write(arr[i] + " ");
        }
        Console.WriteLine("");
    }
}
